native-embedding/src/test/c/nativejavavm.c

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <jni.h>
+
+#define CHECK_THROW() if ((*env)->ExceptionCheck(env)) { \
+    printf("JNI error at %s:%d\n", __FILE__, __LINE__); \
+    exit(1); \
+}
+
+JavaVM* vm = NULL;
+JNIEnv* env = NULL;
+
+jclass t_value = NULL;
+jclass t_api = NULL;
+
+jmethodID m_value_asString = NULL;
+jmethodID m_api_helloWithCallback = NULL;
+jmethodID m_api_fibonacci = NULL;
+jmethodID m_api_eval = NULL;
+jmethodID m_api_setup = NULL;
+jmethodID m_api_tearDown = NULL;
+
+
+const char* get_string_chars(jstring str) {
+   jboolean isCopy;
+   const char* chars = (*env)->GetStringUTFChars(env, str, &isCopy);
+   CHECK_THROW()
+   return chars;
+}
+
+void release_string_chars(jstring str, const char *chars) {
+   (*env)->ReleaseStringUTFChars(env, str, chars);
+   CHECK_THROW()
+}
+
+char *getOptionString(char *optionParam, char *optionValue) {
+    const int optionStringLength = strlen(optionParam) + strlen(optionValue);
+    char *optionString = calloc(optionStringLength + 1, sizeof(char));
+    strcpy(optionString,  optionParam);
+    strcat(optionString, optionValue);
+    return optionString;
+}
+
+void create_java_vm(int argc, char** argv) {
```

Comment:
Add a comment that for Native we can create multiple java ms per process. For HotSpot we cannot.